modules/imgimport/include/csvreading.h

```diff
@@ -0,0 +1,172 @@
+#include <fstream>
+#include <string>
+#include "metadata.h"
+#include <vector>
+#include <iostream>
+using namespace std;
+
+struct Metadatalinkedlist{
+    Metadata a;
+    Metadatalinkedlist* next;
+};
+
+Metadatalinkedlist* readcsv(const char* filename){
```

Comment:
This function should be declared here and defined in a separate source file. This prevents files that include this header from pulling in all of its includes (I think metadata.h is the only include the header itself should need), and also hides implementation details.
You should also include a Doxygen header comment describing the function's purpose, its parameters and its return value.